src/App.js

```diff
@@ -30,23 +31,111 @@ const App = () => {
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
 
+  const [currentSquare, setCurrentSquare] = useState(false);
+
   // Wave 2
+  const changeTheSquare = (id) => {
+    let newSquares = [];
+    for (let row = 0; row < 3; row += 1) {
+      newSquares.push([]);
+      for (let col = 0; col < 3; col += 1) {
+        if (id === squares[row][col].id) {
+          if (!squares[row][col].value) {
+            if (!currentSquare) {
+              squares[row][col].value = PLAYER_1;
+            } else {
+              squares[row][col].value = PLAYER_2;
+            }
+            setCurrentSquare(!currentSquare);
+          }
+        }
+        newSquares[row].push(squares[row][col]);
+      }
+    }
+    setSquares(newSquares);
+  };
+
   // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
   const checkForWinner = () => {
-    // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if
-    //    all three squares have the same value.
+    //check rows
+    for (let row = 0; row < 3; row += 1) {
+      let countX = 0;
+      let countO = 0;
+      for (let col = 0; col < 3; col += 1) {
+        if (squares[row][col] === 'X') {
```

Comment:
This kind of change should be made throughout the function. 

```suggestion
        if (squares[row][col].value === PLAYER_1) {
```